client/a8c-for-agencies/sections/partner-directory/partner-directory.tsx

```diff
@@ -87,6 +97,18 @@ export default function PartnerDirectory( { selectedSection }: Props ) {
 		return sections;
 	}, [ translate, agencyDetailsData, applicationData ] );
 
+	// Wait until the agency is fetched
+	if ( ! hasAgency || isFetching ) {
+		return null;
```

Comment:
Yes, I was going to add that, but as per this comment on Slack and because I saw that a global placeholder is already shown when you access directly to the URL, an empty screen never shows up. So basically that condition is to avoid errors with no agency data fully loaded.